scripts/reducers/flows.reducer.js

```diff
@@ -175,12 +175,15 @@ export default function (state = {}, action) {
       links = mergeLinks(state.unmergedLinks);
     }
 
-    let mapNodeColors = [];
+    let mapNodeColors = [],
+      selectedNodeColors = [];
     if (nodesColoredBySelection && nodesColoredBySelection.length !== 0) {
       //TODO: finish this so that we can color the map too
-      mapNodeColors = getMapColorsFromLinks(links);
+      const colorsFromLinks = getColorsFromLinks(links);
```

Comment:
You can directly use `recolorGroups` I think (collect the values from the dict using `Object.values` for example)
This means you don't need to loop through all links for that, in fact you probably don't even need to add property to the app state, just reuse recolorGroups.
As a side note, `getMapColorsFromLinks` and the `mapNodeColors` on the state are not used. My bad I should have clean that up.